docusaurus/docs/Android/04-compose/07-guides/02-customizing-image-and-video-previews.mdx

```diff
@@ -0,0 +1,166 @@
+                modifier = Modifier
+                    .widthIn(10.dp)
+                    .padding(2.dp)
+                    .background(
+                        color = Color(red = 256, blue = 256, green = 256, alpha = 220),
+                        shape = RoundedCornerShape(8.dp)
+                    )
+                    .fillMaxWidth(0.3f)
+                    .aspectRatio(1.20f),
+            )
+        }
+    },
+    // Render a custom item above attachments inside the message composer
+    previewItemOverlayContent = { attachmentType ->
+        // Apply it only to video attachments
+        if (attachmentType == AttachmentType.VIDEO) {
+            CustomPlayButton(
+                modifier = Modifier
+                    .padding(2.dp)
+                    .background(
+                        color = Color(red = 256, blue = 256, green = 256, alpha = 220),
+                        shape = RoundedCornerShape(8.dp)
+                    )
+                    .fillMaxWidth(0.35f)
+                    .aspectRatio(1.20f),
+            )
+        }
+    })
+```
+
+Since we only want to modify `MediaAttachmentFactory` and keep the rest of the default factories, we'll copy them in the same order they are created in [StreamAttachmentFactories.defaultFactories()](https://github.com/GetStream/stream-chat-android/blob/main/stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/attachments/StreamAttachmentFactories.kt#L65).
+
+```kotlin
+val attachmentFactories = listOf(
+    UploadAttachmentFactory(),
+    LinkAttachmentFactory(linkDescriptionMaxLines = 5),
+    GiphyAttachmentFactory(),
+    customMediaAttachmentFactory,
+    FileAttachmentFactory(),
+    UnsupportedAttachmentFactory()
+)
+```
+
+Remember, order of precedence matters when creating a list of attachment factories, as the first one whose `canHandle()` function evaluates to true will render the given attachment.
+
+Finally, let's replace the default attachment factories:
+
+```kotlin
+override fun onCreate(savedInstanceState: Bundle?, persistentState: PersistableBundle?) {
+    super.onCreate(savedInstanceState, persistentState)
+
+    setContent {
+        // Replace the default attachment factories
+        ChatTheme(attachmentFactories = attachmentFactories) {
+            MessagesScreen(
+                viewModelFactory = messageListViewModelFactory,
+                onBackPressed = { finish() }
+            )
+        }
+    }
+}
+```
+
+And the resulting experience looks like this:
+
+![Custom Image and Video Attachment Previews](../../assets/compose_customizing_image_and_video_attachments_custom.png)
+
+## Conclusion
+
+If you need to put small personal touches to how your app displays image and video attachments while expending very little time on development, the approach we've discussed is optimal. If you need to do large changes, remember you can always create your own attachment factory and replace the default one with it.
```

Comment:
There are guides how to create a custom attachment. Maybe link it somewhere?

Reply:
Added a note below.